### ISSUE TITLE
Are you all ready to make the toolkit work as soon as the web app is reopened ?

### ISSUE THREAD

**Issue by DjTrilogic** [trydis/FIFA-Ultimate-Team-Toolkit#128]
Hello Everyone,

(this is not an issue, it's a poll ^^)

Greeting!

**Comment by Levesque7** [comment]
If it's a captcha on every card before your fist bid on that card, we are smoked. But yes, I would very much be in for getting this working again.

**Comment by ulrickpsp** [comment]
+1

**Comment by artielange** [comment]
I doubt they're adding captchas to bidding. Did you hear that somewhere?

**Comment by ulrickpsp** [comment]
I don't think so either.

**Comment by Levesque7** [comment]
No, I didn't hear it anywhere, and I don't think they will either, I was just saying that's the only way they can truly get rid of auto buyers. 

Who knows if that's even the reason for the apps being taken down.

**Comment by ulrickpsp** [comment]
I think if they wanted to prevent the use of AB's, the WebApp would never be reopened

**Comment by artielange** [comment]
@levesque7 are you French or maybe from Canada?

**Comment by Levesque7** [comment]
@ulrickpsp I agree.

@artielange Both.

**Comment by artielange** [comment]
@Levesque7 Je suis content de voir un autre Quebecois dans la gang !

Nothing we can do but wait I suppose. Could it be they're splitting the markets? Ie XBone and X360, two different markets, same for PS3/PS4 ?

**Comment by ulrickpsp** [comment]
As a PS3 player I hope that's wrong. But this task is taking quite long..so could be a possibility.

**Comment by Levesque7** [comment]
@artielange Je suis Ontarien, mes ancetres etaient Quebecois. (Clavier anglais)

What would be the purpose of them splitting the markets? Is there some kind of advantage for them to doing this? Also, wasn't the app down for 3 months last year? I have no faith that it will be coming back any time soon. What were the changes that affected the toolkit after it came back last year? (I just started using the toolkit a few weeks ago.)

**Comment by artielange** [comment]
I'm not sure what the advantage would be of splitting the markets. It seems like some of their other titles have split markets 

The WebApp was down last year for a lot longer than 3 months, but it was done due to the servers not being able to cope with the strain of bots/regular use. It was an "emergency". It didn't come back until FUT15 lol

This year it's planned. So it's almost certain that it's coming back.

**Comment by ulrickpsp** [comment]
I talked to a EA advisor and he assured me that the web will be back. Of course he gave me no time frame for this.

**Comment by artielange** [comment]
No I didn't read anything but I'm sure they may have tweeted some stuff saying they'd need to do maintenance. I'm surprised they're so public about the maintenance with the banner on the webapp. It's gonna be close to a week soon that it has been down

**Comment by Levesque7** [comment]
@ulrickpsp EA advisors are historically clueless, but I hope you are right. 

I hope it comes back up soon. I was just starting to make some real coin from my app...

**Comment by lucasvarjao** [comment]
I have heard rumors that EA is planning to release an update this weekend, perhaps the webapp come in the same time

**Comment by Yifei022** [comment]
I guess the advantage of splitting market is that the current coin glitch comes from old gen. It's highly possible that there might be more glitches found from old gen if the current glitch gets patched.  Breaking down the generation can help prevent market inflation on the new console if EA does not want to spend too much effort on fixing the glitches on old gen.

**Comment by svickers** [comment]
From the recently released FAQ:

**When is the Web App/Companion App Transfer Market coming back up?**
The Transfer Market will remain inaccessible from Web and Companion Apps


https://www.easports.com/fifa/ultimate-team/news/2015/fut-price-ranges-faq

It was fun while it lasted.

**Comment by trydis** [comment]
:boom:

I'm hoping it'll return in the next iteration, but this time with security in mind. I'm surprised they didn't care at all for so long.

**Comment by craigbaines** [comment]
Ding Ding. 

If anyone wants to pull there efforts together then add me ill start a skype chat - craig_b3

**Comment by Levesque7** [comment]
Hey, I'm so down for getting this working, but I can't help tonight. It doesn't even seem on the surface like anything has changed other than aesthetics.. Keep us updated!

**Comment by craigbaines** [comment]
it seems to log in for me atm, but all my requests such as credits / tradepiles dont work, not too sure why atm lol.

**Comment by DjTrilogic** [comment]
Im so enthusiastic to do it this year, as last year i discovered FUT very late, and couldn't make big deals with my bot (futmarketbot). So this year i'm ready to pull a max. 
My skype : trilogicabdess

--
BENSLIMANE Abdessamad
Téléphone: +33 6 05 56 47 27

**Comment by ulrickpsp** [comment]
I'm in!

**Comment by jonnydowe** [comment]
I've made good progress. Have sorted login and retrieving the trade pile successfully in a couple of hours last night. Onwards today.

**Comment by DjTrilogic** [comment]
Would be great to pull your work :p. I’ll join the club tonight or tomorrow night !

 

De : jonnydowe [mailto:notifications@github.com] 
Envoyé : mercredi 16 septembre 2015 15:03
À : trydis/FIFA-Ultimate-Team-2015-Toolkit <FIFA-Ultimate-Team-2015-Toolkit@noreply.github.com>
Cc : DjTrilogic <abdessamad.benslimane@gmail.com>
Objet : Re: [FIFA-Ultimate-Team-2015-Toolkit] Are you all ready to make the toolkit work as soon as the web app is reopened ? (#128)

 

I've made good progress. Have sorted login and retrieving the trade pile successfully in a couple of hours last night. Onwards today.

**Comment by ulrickpsp** [comment]
How did you manage to log in? As far as I know, verification code is no longer needed, right? I'm getting error in the line:

var mainPageResponseMessage = await HttpClient.GetAsync(Resources.Home);

I've checked that resource and I have this:

public const string Home = "https://www.easports.com/fifa/ultimate-team/web-app";


is it wrong?

**Comment by craigbaines** [comment]
did you change all you other resource urls? i changed them from fifa15 to fifa16 where needed, i can log in but my tradepile requests etc aint working, not sure what ive done wrong yet, not looked back at it since last night. 

jonny, if you could paste a list of the resource urls you've updated it would be super helpful! 

Thanks.

**Comment by kauefelipe** [comment]
If someone makes a "stable" version, make us know! I am ready also to collaborate, I have been developing since the FUT 14 ...

**Comment by ulrickpsp** [comment]
That would be awesome. I'm currently facing that, so I'd save a lot of time to invest in new features

**Comment by trydis** [comment]
I'm not playing Ultimate Team anymore, but i can create a new repo, create NuGet packages etc. if that would be helpful. Once we're certain we can get something working i can create the repo.

**Comment by craigbaines** [comment]
As always trydis it would be more than appreciated by all of us! If i get anything working ill try an throw what i find here to make it easier / quicker for you.

**Comment by jonnydowe** [comment]
Okay here's a copy pasta of my Resouces.cs. I'm concentrating on getting it working for Xbox One, so these might not function so well on other platforms. There are probably other places in your code where you'll need to change URLs as a few are hardcoded, especially in LoginRequest.cs.

        public const string Validate = "https://www.easports.com/iframe/fut16/p/ut/game/fifa16/phishing/validate";

        public const string Auth = "https://www.easports.com/iframe/fut16/p/ut/auth";
        
        public const string LoggedIn = "https://www.easports.com/fifa/api/isUserLoggedIn";

        public const string Home = "https://www.easports.com/fifa/ultimate-team/web-app";

        public const string NucleusId = "https://www.easports.com/iframe/fut16/?locale=en_US&baseShowoffUrl=https%3A%2F%2Fwww.easports.com%2Ffifa%2Fultimate-team%2Fweb-app%2Fshow-off&guest_app_uri=http%3A%2F%2Fwww.easports.com%2Ffifa%2Fultimate-team%2Fweb-app";

        public const string Shards = "https://www.easports.com/iframe/fut16/p/ut/shards/v2?_={0}";

        public const string AccountInfo = " https://www.easports.com/iframe/fut16/p/ut/game/fifa16/user/accountinfo?sku=FUT16WEB&_={0}";

        public const string BaseShowoff = "http://www.easports.com/iframe/fut16/?baseShowoffUrl=https%3A%2F%2Fwww.easports.com%2Fuk%2Ffifa%2Fultimate-team%2Fweb-app%2Fshow-off&guest_app_uri=http%3A%2F%2Fwww.easports.com%2Fuk%2Ffifa%2Fultimate-team%2Fweb-app&locale=en_GB";

        public const string Item = "https://fifa15.content.easports.com/fifa/fltOnlineAssets/B488919F-23B5-497F-9FC0-CACFB38863D0/2016/fut/items/web/{0}.json";

        public const string PlayerImage = "http://fifa15.content.easports.com/fifa/fifa/fltOnlineAssets/B488919F-23B5-497F-9FC0-CACFB38863D0/2016/fut/items/images/players/web/{0}.png";

        public const string FlagsImage = "http://fifa15.content.easports.com/fifa/fltOnlineAssets/B488919F-23B5-497F-9FC0-CACFB38863D0/2016/fut/items/images/cardflagssmall/web/{0}.png";

        public const string ClubImage = "http://fifa15.content.easports.com/fifa/fltOnlineAssets/B488919F-23B5-497F-9FC0-CACFB38863D0/2016/fut/items/images/clubbadges/web/s{0}.png";

        public string FutHome = "https://utas.s3.fut.ea.com/ut/game/fifa16/";

        public const string FutHomeXbox360 = "https://utas.s3.fut.ea.com/ut/game/fifa16/";

        public const string MyClub = "club?level=10";

        public const string TransferMarket = "transfermarket";

        public const string Bid = "trade/{0}/bid";

        public const string TradeStatus = "trade/status?tradeIds={0}";

        public const string Credits = "user/credits";

        public const string Auctionhouse = "auctionhouse";

        public const string TradePile = "tradepile";

        public const string Watchlist = "watchlist";

        public const string PurchasedItems = "purchased/items";

        public const string ListItem = "item";

        public const string QuickSell = "item?itemIds={0}";

        public const string RemoveFromTradePile = "trade/{0}";

        public const string PileSize = "clientdata/pileSize";

        public const string Consumables = "club/stats/consumables";

        public const string SquadList = "squad/list";

        public const string SquadDetails = "squad/{0}";

        public const string ReList = "relist";

**Comment by ulrickpsp** [comment]
If you could share at least a working version with the login, I'd really appreciate it. Every year I have tons of issues regarding the login process. Thanks in advance.

**Comment by craigbaines** [comment]
it looks like the shard bits effected from the new url, its giving multiple "skus" an therefore failing fetching it? 

Im assuming i need to edit my model? Im with you ulr, this bit confuses me too haha, i will post back if i solve this but in the meantime any working bits already done id appreciate the share!

**Comment by craigbaines** [comment]
jonnydowe did you encounter expired session errors? Any solution?

**Comment by ulrickpsp** [comment]
System.TypeInitializationException in FutRequestBase, any idea?

**Comment by craigbaines** [comment]
i had to add         

public string[] skus { get; set; }

into shard.cs

is it that?

**Comment by jonnydowe** [comment]
Hi, yeah from what I remember that was one of the issues (I did that all last night). What I'm now looking for is an updated list of all team IDs. Has anyone found the json for that yet?

**Comment by bradders591** [comment]
I've got the clubs list, gimme a sec

**Comment by bradders591** [comment]
This should work:
http://www.megafileupload.com/9luP/clubs.json

**Comment by jonnydowe** [comment]
That's great thanks, where did you source that from, if I may ask?

**Comment by bradders591** [comment]
They have a player DB search tool on their website, i've grabbed all the data, stuck it in a DB locally and pulled out distinct club values :) give me a shout if you need anything else.

**Comment by jonnydowe** [comment]
Wow that's great work. Thanks! Have you transferred it into the Team.cs file yet? That's the main thing as it's a core part of the API.

**Comment by craigbaines** [comment]
can you guys post your loginrequest.cs?

**Comment by bradders591** [comment]
I haven't added anything to the API toolkit, not had time but can supply any more info if needed

**Comment by craigbaines** [comment]
Just having a bit of a brain fart, been so long since i dabbled in this code lol. Im logging in, getting the auth / shards an what not, its just when i get to tradepile or any other form of request, best i can do is get an expired session error lol.

**Comment by jonnydowe** [comment]
Do a CTRL+F for utas and make sure you're accessing utas.s3.fut.ea.com with every request. s3 appears to be the address used for the current-gen consoles. That should fix your expired sessions.

**Comment by craigbaines** [comment]
yeah i did check that, and its right. 

I actually set the club account up using 360, would that be any different?

**Comment by jonnydowe** [comment]
Yeah I reckon that might be your issue, as s3 is only for XBONE/PS4 accounts.

Has anyone got the Nation and League ids updated cs file?

**Comment by jonnydowe** [comment]
This page has all the league, club etc IDs along with their English names. Just need to parse it. Anyone able to? 

https://www.easports.com/iframe/fut16/bundles/futweb/web/flash/xml/localization/messages.en_GB.xml?cl=154151

**Comment by lucasvarjao** [comment]
This json file has all the league,club, nationality, chemistry styles and many others things that i got it from the EA database using fiddler

http://www.megafileupload.com/9lxH/types.json

**Comment by craigbaines** [comment]
Thing is though im looking at fiddler on the web app and its using s3 too :/

**Comment by craigbaines** [comment]
within my login request i change dteh s2 to s3 here:

var route = string.Format("https://utas.{0}fut.ea.com:443", platform == Platform.Xbox360 ? string.Empty : "s3.");

Just realised that its not working unless i actually put it in like: 

 var route = string.Format("https://utas.s3.fut.ea.com:443", platform == Platform.Xbox360 ? string.Empty : "s3.");

That solved my session expired if anyones etting stuck there.

**Comment by DjTrilogic** [comment]
My bad luck ! Tomorrow is my last day in university and I have to work on my final presentation... I'm soooo impatient to finally open my visual studio and make this shit work... 
Hopefully by tomorrow you guys can make a new repo to restructure the work.

--
BENSLIMANE Abdessamad
Téléphone: +33 6 05 56 47 27

**Comment by craigbaines** [comment]
Am i right in saying resource id's are negative now? 

uint's wont work ?

**Comment by ulrickpsp** [comment]
Would it be possible to have the updated code until the login? So I can help with any other tasks.. :)

**Comment by craigbaines** [comment]
i found that s2 seems to work for ps and s3 seems to work for xbox, regardless of what console?

So i guess just replace all utas urls with a .s2 or .s3, an use the resources from above. Thats literally all ive done an it logs in ok for me, hopefully someone throws a more elegant solution together, i would share mine but ive modded it a fair bit and i cant remember which bits are changed and are not, it would be more confusion than its worth to everyone lol

**Comment by craigbaines** [comment]
Is anyone able to comment on the resource id value? 

I make my searches for in forms based on the whole definition id thing, which calculates resource id from asset id etc etc...

Using the same calcs it seems to work, but part of my code checks the webapp cards resource id against my own and there obviously different, web app values all seem to be negative now?

**Comment by ulrickpsp** [comment]
What changes have you done in the code just for being able to log in? Always getting "Unable to login".   That would be a helpful tip. One even more helpful would be to point me the steps you've done so I can learn from this. Thanks a lot.

**Comment by bradders591** [comment]
anyone come across this when trying to login?:

<code>phishingToken:  ---> System.IndexOutOfRangeException: Index was outside the bounds of the array.   at UltimateTeam.Toolkit.Requests.LoginRequest.<GetSessionIdAsync>d__11.MoveNe
xt()</code>

**Comment by meshoe** [comment]
What I did to make the log in work:

- changed Resources.cs to point to fut16 like jonnydowe showed above
- added public string[] Skus { get; set; }  at the end of Shard.cs
- changed FUT address according to platform wherever it appears in LoginRequest.cs (except GetShardsAsync()):
             * utas.s3.fut.ea.com:443 should be used for XBOX 360/One
             * utas.s2.fut.ea.com:443 should be used for PS3/PS4/PC
             * utas.fut.ea.com:443 remains for various Fifa14/15 platforms

Most of this was already explained above, just put it together. Now I'm able to login successfully on PC FUT.

**Comment by hfielding** [comment]
As a further note to the above post, within the GetSessionIdAsync method of the LoginRequest class, people who have multiple personas/consoles are going to need/want to add the 'gameSku' param to the JSON post.

"FFA16XBX" - Xbox 360,
"FFA16XBO" - Xbox One.

I do not have the playstation or PC values as I only have these two consoles.

**Comment by meshoe** [comment]
FFA16PS3 - PS3
FFA16PS4 - PS4
FFA16PCC - PC

**Comment by craigbaines** [comment]
What does adding that achieve?

**Comment by hfielding** [comment]
It will just allow you to switch between your Playstation 3/4 and Xbox 360/one clubs.

I don't know if the parameter is compulsory but, if we can mirror the web app json completely then it's less avenues for us to be detected.

**Comment by craigbaines** [comment]
So for example, you saying to tweak it to this? :

string.Format(@"{{ ""isReadOnly"": false, ""sku"": ""FUT16WEB"",  ""gameSku"": ""FFA16XBX"",  ""clientVersion"": 1, ""nucleusPersonaId"": {0}, ""nucleusPersonaDisplayName"": ""{1}"", ""nucleusPersonaPlatform"": ""{2}"", ""locale"": ""en-GB"", ""method"": ""authcode"", ""priorityLevel"":4, ""identification"": {{ ""authCode"": """" }} }}",

**Comment by craigbaines** [comment]
Correct me if im wrong but this should add in what you suggested right? 

Not much of a share but here you guys go... 

Swap inside the loginrequest: 

private async Task<string> GetSessionIdAsync(UserAccounts userAccounts, Platform platform)
        {
            var persona = userAccounts
                .UserAccountInfo
                .Personas
                .LastOrDefault(p => p.UserClubList.Any(club => club.Platform == GetNucleusPersonaPlatform(platform)));
            if (persona == null)
            {
                throw new FutException("Couldn't find a persona matching the selected platform");
            }
            var authResponseMessage = await HttpClient.PostAsync(Resources.Auth, new StringContent(
               string.Format(@"{{ ""isReadOnly"": false, ""sku"": ""FUT16WEB"", ""clientVersion"": 1, ""nucleusPersonaId"": {0}, ""nucleusPersonaDisplayName"": ""{1}"", ""nucleusPersonaPlatform"": ""{2}"",""gameSku"": ""{3}"", ""locale"": ""en-GB"", ""method"": ""authcode"", ""priorityLevel"":4, ""identification"": {{ ""authCode"": """" }} }}",
                    persona.PersonaId, persona.PersonaName, GetNucleusPersonaPlatform(platform), GetgameSkuPlatform(platform))));
            authResponseMessage.EnsureSuccessStatusCode();
            var sessionId = Regex.Match(await authResponseMessage.Content.ReadAsStringAsync(), "\"sid\":\"\\S+\"")
                .Value
                .Split(new[] { ':' })[1]
                .Replace("\"", string.Empty);

            return sessionId;
        }

        private static string GetgameSkuPlatform(Platform platform)
        {
            switch (platform)
            {
                case Platform.Ps4:
                    return "FFA16PS4";
                case Platform.XboxOne:
                    return "FFA16XBO";
                case Platform.Ps3:
                    return "FFA16PS3";
                case Platform.Xbox360:
                    return "FFA16XBX";
                case Platform.Pc:
                    return "FFA16PCC";
                default:
                    throw new ArgumentOutOfRangeException("platform");
            }
        }

And inside platform.cs model add

        XboxOne,
        Ps4,


Has anyone come up with an elegant solution to being able to switch the fut home resource depending on the platform? 

the parts inside the login request such as GetUserAccountsAsync ive changed over this lien to the following: 

    var route = string.Format("https://utas.{0}fut.ea.com:443", platform == Platform.Xbox360 ? "s3." : platform == Platform.XboxOne ? "s3." : platform == Platform.Ps3 ? "s2." : platform == Platform.Ps4 ? "s2." : string.Empty);
          

Which seems to do the trick

**Comment by bradders591** [comment]
Cheers for that Craig, seems to be similar to what i have. Login seems to be working fine for me, just something as simple as getting credits is now failing! Granted its my old code using this DLL but i don't think much else has changed!

**Comment by craigbaines** [comment]
I had the same issue, make sure the futhome resource link has .s3 included in it, or .s2 for ps, im trying to figiure out a way now to make it so that it switches but im a little stumped, any suggestions?

**Comment by bradders591** [comment]
futhome has s3... I'll keep digging.

Yeah, i'm the same as you with this. I'll have a think!

**Comment by craigbaines** [comment]
This is probably a horrible way to do it but! 

inside the resource class ive added to the top: 

    internal class Resources
    {

        static string FutHomeset;
        static public async void setPlatform(string plat)
        {
            if (plat == "Xbox360")
            {
                  FutHomeset = "https://utas.s3.fut.ea.com/ut/game/fifa16/";     
            }
            else if (plat == "XboxOne")
            {
                FutHomeset = "https://utas.s3.fut.ea.com/ut/game/fifa16/";
            }
            else if (plat == "Ps3")
            {
                FutHomeset = "https://utas.s2.fut.ea.com/ut/game/fifa16/";
            }
            else if (plat == "Ps4")
            {
                FutHomeset = "https://utas.s2.fut.ea.com/ut/game/fifa16/";
            }
            else if (plat == "PC")
            {
                FutHomeset = "https://utas.s2.fut.ea.com/ut/game/fifa16/";
            }
        }

        public string FutHome =  FutHomeset;



Add a reference to the tool kit resources then i just run something like this just before i start a new instance of the client. 
 
Resources.setPlatform("Xbox360"); 
...or whatever platform your using. I grab my string from a platform dropdown menu at login. 

Its fugly an probably the worse way of doing it but it does seem to work for me so far so hopefully someone can improve on or use this if your really stuck.

**Comment by ulrickpsp** [comment]
Thank a lot to all of you. You're awesome. I'll try to put everything together and see if I can login. Thanks!

**Comment by bradders591** [comment]
Mine was either invalid credentials or a locked account from past experiences of login issues

**Comment by craigbaines** [comment]
Anyone found a solution to the whole "error" thing happening with the transfer market search, only happens when my bots been running for roughly 40 mins +. Same thing as last year when it would "bug" your account, wait long enough it would start working again, same thing but it only seems to be effecting TM searches now. Seems to happen despite what RPM i set,

Anyone got a suggested RPM to use thats working for them?

**Comment by bradders591** [comment]
I haven't! Still not got past my getCredits issue! My RPM for transfer searches is a random number between 10000 and 20000 milliseconds. I only use mine personally so I just do what works for me

**Comment by craigbaines** [comment]
Last year i just made it rotate through accounts, List 1 would run for x mins, then log out an list 2 logged in etc etc. 

I managed to avoid the accounts "bugging" by doing that and running for a time under what seemed to cause the bug. 

But its not ideal, i seriously have no idea how or why these bugs happen though, someone must have an idea? solution? even a hint... lol !?

**Comment by DjTrilogic** [comment]
I used to shuffle between accounts after short running sessions and I never had an account bugged ; Maybe there’s a system in ea server that detect bots no matter how low is your rpm

 

De : craigbaines [mailto:notifications@github.com] 
Envoyé : vendredi 18 septembre 2015 17:18
À : trydis/FIFA-Ultimate-Team-2015-Toolkit <FIFA-Ultimate-Team-2015-Toolkit@noreply.github.com>
Cc : DjTrilogic <abdessamad.benslimane@gmail.com>
Objet : Re: [FIFA-Ultimate-Team-2015-Toolkit] Are you all ready to make the toolkit work as soon as the web app is reopened ? (#128)

 

Last year i just made it rotate through accounts, List 1 would run for x mins, then log out an list 2 logged in etc etc. 

I managed to avoid the accounts "bugging" by doing that and running for a time under what seemed to cause the bug. 

But its not ideal, i seriously have no idea how or why these bugs happen though, someone must have an idea? solution? even a hint... lol !?

**Comment by craigbaines** [comment]
Yep, thats what i mean, if you switch accounts before it gets chance to flag it or whatever they seemed to be fine. 

Id just love to know how or why its being detected.

**Comment by jonnydowe** [comment]
Yeah I'd love to know too. Have you experimented with the keepalive calls being made every 5 minutes by the web app? Perhaps implementing those could help to solve it?

**Comment by craigbaines** [comment]
I dont think i have that, Im pretty sure i spoke to someone who had tried it last year an they said it made no difference, its something that did cross my mind but why i didn't i cant exactly remember.

**Comment by DjTrilogic** [comment]
Probably they calculate the average time between two requests made by a the « suspect » bot and compare it to human player behavior; if we put these data in a chart you’ll see that the bot curve is kind of constant over time … 

 

De : craigbaines [mailto:notifications@github.com] 
Envoyé : vendredi 18 septembre 2015 17:27
À : trydis/FIFA-Ultimate-Team-2015-Toolkit <FIFA-Ultimate-Team-2015-Toolkit@noreply.github.com>
Cc : DjTrilogic <abdessamad.benslimane@gmail.com>
Objet : Re: [FIFA-Ultimate-Team-2015-Toolkit] Are you all ready to make the toolkit work as soon as the web app is reopened ? (#128)

 

Yep, thats what i mean, if you switch accounts before it gets chance to flag it or whatever they seemed to be fine. 

Id just love to know how or why its being detected.

**Comment by craigbaines** [comment]
I guesstimated it must be something to do with an average rpm over a set time.... 

it would make sense why logging out before a certain time would avoid it. 

Other than something along the lines of that logic im stumped as to what it could be. Rotating through accounts is going to be a pain this year too with the 10 account limit... .

That being said, i THINK your actually able to log the same email in twice at teh same time, one for each console weather it be ps or xbox. Mine didn't seem to d/c each other, but i need to log into an xbox one to unlock them on them it says, but im hoping that in essence means we actually have 20 accounts. ill probably rotate my lists to switch between the consoles an keep every account on at once.

**Comment by jonnydowe** [comment]
Yeah average RPM sounds like a likely thing. But how would you counter that?

**Comment by Levesque7** [comment]
Pause operation for 5 minutes every 30?

**Comment by DjTrilogic** [comment]
Run 10~150 minutes and stop 10~50 minutes

 

De : Levesque7 [mailto:notifications@github.com] 
Envoyé : vendredi 18 septembre 2015 17:51
À : trydis/FIFA-Ultimate-Team-2015-Toolkit <FIFA-Ultimate-Team-2015-Toolkit@noreply.github.com>
Cc : DjTrilogic <abdessamad.benslimane@gmail.com>
Objet : Re: [FIFA-Ultimate-Team-2015-Toolkit] Are you all ready to make the toolkit work as soon as the web app is reopened ? (#128)

 

Pause operation for 5 minutes every 30?

**Comment by craigbaines** [comment]
Like i said, i made it so i set up account "lists"

set a time for how oftern to switch lists, an set how many lists to rotate throught. 

list one runs for x mins, then log out an list 2 start, then three, four an so on....

You could pause for x mins, but i figured rotating accounts keep me on-line 24/7 minus the logging in times. Bit annoying having to split resources over more accounts, but i guess this year we will be forced to using only a few anyway. 

As i said, im probably going to have the same accounts on each list, but one list will be logging into xbox one, the other xbox 360. Hopefully there treated like two separate things, seemed like it to me on my brief test.

**Comment by Levesque7** [comment]
What did you guys put for the "return" for Platform.XboxOne and Platform.Ps4 in GetNucleusPersonalPlatform in LoginRequest.cs?

**Comment by A-VORONKIN** [comment]
trydis, when new repo will be created?

**Comment by meshoe** [comment]
I see that our toolkit passes the timestamp to each request by appending "?_=<timestamp>". From fiddler I see that the webapp uses an initial timestamp (probably taken from login) and then it increases it by 1 for each request, so it's not sending the real timestamp, but a relative one. Maybe that's why you're getting session expired so often, Craig?

![ut_fiddler](https://cloud.githubusercontent.com/assets/14335663/9977629/3d0d74a2-5f17-11e5-94cd-06fa56c7b0f7.png)

The webapp also creates 2 additional requests:
- once every 5 minutes: www.easports.com/fifa/api/keepalive
- once every 10 minutes: www.easports.com/iframe/fut16/noop?_=<timestamp>

Maybe we should add these to the toolkit too and see if we can keep the connection up for longer.

**Comment by jonnydowe** [comment]
Yeah that seems a good idea, thanks. Have you managed to implement that?

**Comment by trydis** [comment]
I'll try to take a look at the changes tonight, haven't looked at anything yet. I'll continue to use this repo, just renamed it to Ultimate Team Toolkit

**Comment by craigbaines** [comment]
Guys, regards the resource ids being sent back in the json, i grabbed three random guys an calculated the difference between the negative number there  giving us and the actual resource id was  4026531840 .

but i just found a player whos is 3976200192. So im assuming there must be some sort of variable, has anyone any clue / idea how or why the resource id is now negative, an how to convert it to an actual resource id?

**Comment by craigbaines** [comment]
Hows this look to you, timings were in seconds rather than min though. 

I assume nothing else needs the timestamp adding too it? only the repeating noop calls?

![7d78ea130e0c0b408d9f22cb4b9a8ab8 1](https://cloud.githubusercontent.com/assets/9383958/9982483/254fa636-5fd6-11e5-8c0d-2dccb62bc7a9.png)

**Comment by trydis** [comment]
When logging in, the password isn't sent in plain text anymore. If your password is `abc123` it will use a hashed/encrypted value. How did you guys solve this? 

If you enter an incorrect password in the web app and get the incorrect password message it starts posting plain text password like in the earlier editions. After this, it even fails to authenticate with the correct password, since it's sent as plain text.

Example:
`email=some@email.com&password=abc123&_rememberMe=on&rememberMe=on&_eventId=submit&gCaptchaResponse=`

**Comment by craigbaines** [comment]
Didn't touch it, been logging in just fine? lol

**Comment by trydis** [comment]
So you haven't really been logging in i guess, but were already logged in through the web app. My guess is that if you clear your EA cookies and attempt a login through the toolkit, it won't work.

**Comment by bradders591** [comment]
That would probably explain the issues i've been having.

**Comment by craigbaines** [comment]
cleared my cookies via chrome. 

Cleared cookies on bot.

Logged in fine, its not even asking me for two step codes? 

Hate to be adding confusion here but yeah, thats how it is lolol! ?

Let me know if you want me to try / test anything else.

**Comment by trydis** [comment]
Confusing indeed. Can some of you please check the request data in Fiddler, when logging in using the web app. The POST request to https://signin.ea.com/p/web/login and check whether your password is sent in plain text or not.

I'm not very keen on making a NuGet package etc. until i know it's working ok. On the other hand, it's not that hard to circumvent the hashing/encryption/obfuscation of the password, even if we don't figure out the algorithm. Not very user friendly, but we could just grab the value from Fiddler and use that value as our password with the toolkit.

**Comment by craigbaines** [comment]
Plain text here.....

**Comment by trydis** [comment]
Weird. Still voodoo magic value for me, but the toolkit login process works when i use that value.

**Comment by craigbaines** [comment]
so strange! 

have you any thoughts on the resource id now? When i get the tradepile, he items resource id's are all negative, i thought i cracked it by adding a specific number but so far ive come across one guy that doesn't apply so im sure there are others?

**Comment by trydis** [comment]
No, sorry. Played 2 demo matches and that's about it :smile:

**Comment by craigbaines** [comment]
iv'e not even done that lol!

**Comment by Levesque7** [comment]
I can confirm toolkit also works for me with plaintext password. Is it a new account @trydis? Or perhaps a security setting in the Origin profile?

**Comment by trydis** [comment]
@Levesque7 No, several years old. I tried turning off "login verification", but didn't make a difference.

**Comment by jonnydowe** [comment]
I too am using plaintext password on multiple accounts.

Secondly, is there any way of finding out the position of a player? The player's data json (ie https://fifa15.content.easports.com/fifa/fltOnlineAssets/B488919F-23B5-497F-9FC0-CACFB38863D0/2016/fut/items/web/182495.json) does not contain any data which I can see that would tell me this.

Thanks!

**Comment by trydis** [comment]
So i tried with another old account, which i haven't used in years, and that one sends the plain-text password. No idea why my main account triggers some weird password massaging on their servers. 

But at least it will work ok for most people, so i guess i can merge the stuff into the master branch and create the NuGet package. I'll do that tomorrow.

Also, i'm going to close this thead once that's done, since it's getting very long and hard to follow. So please create new issues for each question/issue you're having.

**Comment by craigbaines** [comment]
im curious, on the account with the voodoo magic, does using plain text passwords still log in ok or not?

**Comment by Gamer3** [comment]
Has anyone here managed to get hold of complete player information for FUT 16? Similar to Fut head

**Comment by Levesque7** [comment]
Can anyone help me with how to use the position change search parameters? I want to search for a specific position change type, but I'm unsure of how to enter the parameter properly for that purpose. I only need to know how to define the specific position change type.

**Comment by bradders591** [comment]
This should be an extract of all the players, most the info is there (i think i stripped some stuff out i didn't need before i put it in my DB). Just exported it from my own mongo database so its in JSON format. Had to tar and gzip the file as its ~35mb uncompressed.

http://www.filedropper.com/playerstar

**Comment by Gamer3** [comment]
Hi Bradders591 is that for Fifa 15 or 16?

**Comment by bradders591** [comment]
Should be 16

**Comment by Gamer3** [comment]
I just notice the head snapshots says "https://fifa15.content.easports.com/fifa/fltOnlineAssets/B488919F-23B5-497F-9FC0-CACFB38863D0/2016/fut/items/images/players/html5/120x120/180289.png"

Is there a way to get new fifa 16 images or are they the same?

**Comment by bradders591** [comment]
Seem to be the same.

**Comment by Gamer3** [comment]
Bradders591 do you have Skype or an email I could contact you over please?

**Comment by trydis** [comment]
@craigbaines Nope, doesn't work with that account.

**Comment by bradders591** [comment]
Skype username is Bradders591

**Comment by jonnydowe** [comment]
@Bradders591 How did you get all this data? Where did you source it from?

**Comment by craigbaines** [comment]
I know you plan to close this thread but i just wanted to write here to sum up my questions above, in case anyone else has the same problem! 

The negative resource id seems to be the same lodgic, just add 4026531840 to it to get the real resource id, the problems i had were all players that infact were version 2's (summer transfers) and i was treating them as base cards. 

Once i used there actual resource id, it worked the same way, + 4026531840 .

**Comment by xtheokles** [comment]
Hey guys, how did you solved the error when getting the TradePile? I'm getting this error:  "auctionInfo[0].coinsProcessed', line 1, position 1262.".

**Comment by trydis** [comment]
@xtheokles Have you ever seen a value in the `tradeOwner` field? I noticed you chose `bool` in the pull request. If it may be `null`, `true` or `false` you should set it to `bool?` (nullable boolean).

**Comment by DjTrilogic** [comment]
In order to structure the repository, i'll close this issue. Please open an issue for each "issue" !
